### PR TITLE
ABQM plugin

### PR DESCRIPTION
**Pull request by ibaldin**
Be sure to approve PR on FIM repo and use FIM 0.38. 

This is a functioning plugin that is ready for testing. The code in plugin has  DEBUG_FLAG, when set to true does not try to talk to the actor database and produces an Aggregated BQM like everything is available. I was not able to test without this flag, moderate confidence that it works. All database access is in a single function called 
```
def __occupied_node_capacity(self, *, node_id: str) -> Tuple[Capacities,
                                                                 Dict[ComponentType, Dict[str, Capacities]]]:
```
which produces a tuple of a capacity object of all allocated capacities for this server and a dictionary organized by component resource type/model of capacities taken up on components within this node. 

To test do something like this (replace reading model from file with reading it from API call):

```
import fim.user as fu
Backend MacOSX is interactive backend. Turning interactive mode on.
topo = fu.AdvertisedTopology(graph_file='abqm.graphml')
print(topo)
RENC: { cpu: 6/6, core: 96/96, ram: 1536/1536G, disk: 109600/109600G, unit: 3/3, }
	Components:
		GPU-RTX6000:  GPU RTX6000 { unit: 2/2, }
		GPU-Tesla T4:  GPU Tesla T4 { unit: 4/4, }
		NVME-P4510:  NVME P4510 { disk: 10000/10000G, unit: 10/10, }
		SharedNIC-ConnectX-6:  SharedNIC ConnectX-6 { unit: 3/3, }
		SmartNIC-ConnectX-6:  SmartNIC ConnectX-6 { unit: 2/2, }
		SmartNIC-ConnectX-5:  SmartNIC ConnectX-5 { unit: 2/2, }
	Site Interfaces:
		RENC_UKY: TrunkPort { bw: 100/100G, }
		RENC_LBNL: TrunkPort { bw: 100/100G, }
UKY: { cpu: 6/6, core: 96/96, ram: 1536/1536G, disk: 109600/109600G, unit: 3/3, }
	Components:
		SmartNIC-ConnectX-5:  SmartNIC ConnectX-5 { unit: 2/2, }
		GPU-RTX6000:  GPU RTX6000 { unit: 2/2, }
		GPU-Tesla T4:  GPU Tesla T4 { unit: 4/4, }
		SharedNIC-ConnectX-6:  SharedNIC ConnectX-6 { unit: 3/3, }
		NVME-P4510:  NVME P4510 { disk: 10000/10000G, unit: 10/10, }
		SmartNIC-ConnectX-6:  SmartNIC ConnectX-6 { unit: 2/2, }
	Site Interfaces:
		UKY_LBNL: TrunkPort { bw: 100/100G, }
		UKY_RENC: TrunkPort { bw: 100/100G, }
LBNL: { cpu: 6/6, core: 96/96, ram: 1536/1536G, disk: 109600/109600G, unit: 3/3, }
	Components:
		GPU-RTX6000:  GPU RTX6000 { unit: 2/2, }
		GPU-Tesla T4:  GPU Tesla T4 { unit: 4/4, }
		SharedNIC-ConnectX-6:  SharedNIC ConnectX-6 { unit: 3/3, }
		NVME-P4510:  NVME P4510 { disk: 10000/10000G, unit: 10/10, }
		SmartNIC-ConnectX-6:  SmartNIC ConnectX-6 { unit: 2/2, }
		SmartNIC-ConnectX-5:  SmartNIC ConnectX-5 { unit: 2/2, }
	Site Interfaces:
		LBNL_RENC: TrunkPort { bw: 100/100G, }
		LBNL_UKY: TrunkPort { bw: 100/100G, }
Links:
	l1[L2Path]: ['UKY_RENC', 'RENC_UKY']
	l2[L2Path]: ['UKY_LBNL', 'LBNL_UKY']
	l3[L2Path]: ['LBNL_RENC', 'RENC_LBNL']

```

### CODE REVIEW

File: fabric_cf/actor/fim/plugins/broker/aggregate_bqm_plugin.py [hunk 1]
```diff
@@ -34,20 +35,70 @@
...
+
+                    if allocated_sliver.attached_components_info is not None:
+                        for allocated_component in allocated_sliver.attached_components_info.devices.values():
+                            rt = allocated_component.resource_type
+                            rm = allocated_component.resource_model
+                            if occupied_component_capacities[rt].get(rm, None) is None:
+                                occupied_component_capacities[rt][rm] = Capacities()
+
+                            occupied_component_capacities[rt][rm] = occupied_component_capacities[rt][rm] + \
+                                                          allocated_component.capacities
```
Comment (kthare10) [inline fabric_cf/actor/fim/plugins/broker/aggregate_bqm_plugin.py:99]: allocated_component.capacities should be changed to allocated_component.capacity_allocations

**Comment by kthare10** [comment]
Test Scenario with RENC Ad:
Slice created with following:
```
n1 = t.add_node(name='n1', site='RENC')
cap = fu.Capacities()
cap.set_fields(core=4, ram=64, disk=500)
n1.set_properties(capacities=cap, image_type='qcow2', image_ref='default_centos_8')
n1.add_component(ctype=fu.ComponentType.SmartNIC, model='ConnectX-6', name='nic1')

n2 = t.add_node(name='n2', site='RENC')
n2.set_properties(capacities=cap, image_type='qcow2', image_ref='default_centos_8')
n2.add_component(ctype=fu.ComponentType.GPU, model='Tesla T4', name='nic2')
```
List Resources post slice creation shows correct values for Node capacities but Component capacities are not shown correctly. I think the issue is in the FIM abc_property_graph::component_sliver_to_graph_properties_dict function. This function does not add capacity and label allocation fields to the graph.
```
RENC: { cpu: 6/6, core: 88/96, ram: 1408/1536G, disk: 108600/109600G, unit: 3/3, }
	Components:
		NVME-P4510:  NVME P4510 { disk: 10000/10000G, unit: 10/10, }
		GPU-RTX6000:  GPU RTX6000 { unit: 2/2, }
		GPU-Tesla T4:  GPU Tesla T4 { unit: 4/4, }
		SharedNIC-ConnectX-6:  SharedNIC ConnectX-6 { unit: 3/3, }
		SmartNIC-ConnectX-6:  SmartNIC ConnectX-6 { unit: 2/2, }
		SmartNIC-ConnectX-5:  SmartNIC ConnectX-5 { unit: 2/2, }
	Site Interfaces:
Links:
```

File: fabric_cf/actor/fim/plugins/broker/aggregate_bqm_plugin.py (hunk 1 above)
Comment (ibaldin) [reply fabric_cf/actor/fim/plugins/broker/aggregate_bqm_plugin.py:99]: This depends on how we use the fields. If we already use capacity_allocations, then you are correct

Comment (ibaldin) [reply fabric_cf/actor/fim/plugins/broker/aggregate_bqm_plugin.py]: I fixed this in the following commit

**Comment by ibaldin** [comment]
Note that I removed management mac address field from NetworkNode in FIM - it looks like it was only used in the null handler and anything like that I think we should accommodate via labels.

**Review by kthare10 — APPROVED** [review]

**Pull request merged**